mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -648,4 +648,52 @@ mod tests {
 
         assert_eq!(expected_user_agent, user_agent_header_value);
     }
+
+    #[test]
+    #[ignore]
+    // Simple test to ensure the user agent header is correct even when given profile contains calid credentials.
+    fn test_client_with_valid_credentials() {
+        let expected_user_agent = "mountpoint-s3-client";
+        let profile_name_override = String::from("VALID_CREDENTIALS");
```

Comment:
I am not sure if there are good ways to test remote and automatically:
* configure assume role and sts token on remote testing machine.
* configure `VALIDE_PROFILE` on remote testing machine.